client/src/main/java/org/apache/celeborn/client/read/DfsPartitionReader.java

```diff
@@ -77,10 +78,18 @@ public DfsPartitionReader(
       try {
         TransportClient client =
             clientFactory.createClient(location.getHost(), location.getFetchPort());
-        OpenStream openBlocks =
-            new OpenStream(shuffleKey, location.getFileName(), startMapIndex, endMapIndex);
-        ByteBuffer response = client.sendRpcSync(openBlocks.toByteBuffer(), fetchTimeoutMs);
-        Message.decode(response);
+        TransportMessage openStream =
+            new TransportMessage(
+                MessageType.OPEN_STREAM,
+                PbOpenStream.newBuilder()
+                    .setShuffleKey(shuffleKey)
+                    .setFileName(location.getFileName())
+                    .setStartIndex(startMapIndex)
+                    .setEndIndex(endMapIndex)
+                    .build()
+                    .toByteArray());
+        ByteBuffer response = client.sendRpcSync(openStream.toByteBuffer(), fetchTimeoutMs);
```

Comment:
Are these lines necessary? Seems the response is not used. Is there server-side logic?

Reply:
It's necessary, this means that this file needs to sort.
`if (endMapIndex != Integer.MAX_VALUE) {`